components/content-service-api/initializer.proto

```diff
@@ -12,6 +12,7 @@ message WorkspaceInitializer {
         SnapshotInitializer snapshot = 3;
         PrebuildInitializer prebuild = 4;
     }
+    AdditionalContentInitializer additional_content = 5;
```

Comment:
I'd still much prefer introducing this as an orthogonal concept, something like:

```
message WorkspaceInitializer {
    oneof spec {
        EmptyInitializer empty = 1;
        GitInitializer git = 2;
        SnapshotInitializer snapshot = 3;
        PrebuildInitializer prebuild = 4;
        CompositeInitializer composite = 5;
        FileDownloadInitializer download = 6;
    }
}

// CompositeInitializer uses a collection of initializer to produce workspace content.
// All initializer are executed in the order they're provided.
message CompositeInitializer {
    repeated WorkspaceInitializer initializer = 1;
}

// FileDownloadInitializer downloads files and uses them as workspace content.
message FileDownloadInitializer {
    message FileInfo {
       string url = 1;
       // target_location is relative to the workspace root, e.g. if the workspace is initialized in `/workspace`
       // a target_location of `foobar/file` would produce a file in `/workspace/foobar/file`.
       // target_location must include the filename. The FileDownloadInitializer will create any parent directories
       // necessary to place the file.
       string target_location = 2;
       // digest is a hash of the file content in the OCI digest format (see https://github.com/opencontainers/image-spec/blob/master/descriptor.md#digests).
       // This information is used to compute subsequent
       // content versions, and to validate the file content was downloaded correctly.
       string digest = 3;
    }
    repeated FileInfo files = 1;
}
```

Reply:
(note: integrating the digest mechanism is straight forward using https://github.com/opencontainers/go-digest)